### DISCUSSION TITLE
APCA color contrast checker in Accessibility addon

### DISCUSSION THREAD

**Discussion by acalvino4** [storybookjs/storybook#26008] [Ideas]
### Is your feature request related to a problem? Please describe.

Currently the Accessibility addon checks color contrast using the WCAG 2 recommended ratios. While a good start at a standard, the ratio method has many problems documented. WCAG 3 will recommend the APCA method instead. (Although WCAG 3 is obviously not out yet, the APCA method itself is essentially stable, and given that accessibility is ultimately about user experience, not passing rule sets, I don't think implementation should have to wait on WCAG 3.)

References:
https://github.com/w3c/wcag3/issues/29 (Contains further links to every facet of information on this topic)
https://www.myndex.com/APCA/

### Describe the solution you'd like

Would like to see the APCA method of checking color contrast applied instead of the current 3:1 / 4.5:1 checks (or at least a config option to do so).

### Are you able to assist to bring the feature to reality?

no

**Comment by acalvino4** [comment]
After looking into storybook's source code, I see you are using axe for accessibility tests. Unfortunately, for the time being at least, it [doesn't look like they'll be implementing apca checks](https://github.com/dequelabs/axe-core/issues/3325). On the other hand, Stack Overflow has [published a package](https://github.com/StackExchange/apca-check) that seems to extend axe to provide these tests.

**Comment by peterlaxalt** [comment]
+1 To this
